### PR TITLE
Log exceptions as telemetry for AppDesigner

### PR DESCRIPTION
**Pull request by mavasani**
See https://github.com/dotnet/roslyn-project-system/pull/274 for more context.

### CODE REVIEW

**Comment by mavasani** [comment]
Tagging @dotnet/project-system for review

File: src/Microsoft.VisualStudio.AppDesigner/Common/Utils.vb [hunk 1]
```diff
@@ -242,6 +241,48 @@ Namespace Microsoft.VisualStudio.Editors.AppDesCommon
...
+        ''' <param name="ex">The exception to log and check if it is recoverable.</param>
+        ''' <param name="exceptionEventDescription">Additional description for the cause of the exception.</param>
+        ''' <param name="throwingComponentName">Name of the component that threw that exception, generally the containing type name.</param>
+        ''' <param name="debugFail">If True, then invoke Debug.Fail for recoverable exception.</param>
+        ''' <param name="debugWriteLine">If True, then invoke Debug.WriteLine for recoverable exception.</param>
+        ''' <param name="considerExceptionAsRecoverable">If True, then the given exception is always considered recoverable.
+        ''' Otherwise, invokes <see cref="IsUnrecoverable(Exception, Boolean)"/> to determine if exception is recoverable or not.</param>
+        ''' <param name="ignoreOutOfMemory">If True, out of memory will be considered recoverable.</param>
+        ''' <remarks></remarks>
+        Public Function ReportWithoutCrash(ByVal ex As Exception,
```
Comment (mavasani) [inline src/Microsoft.VisualStudio.AppDesigner/Common/Utils.vb]: This is same code as the one added in https://github.com/dotnet/roslyn-project-system/pull/274. I'll move this into a common file shared by both projects in a follow up change.

**Comment by davkean** [comment]
Your build is broken.

File: src/Microsoft.VisualStudio.AppDesigner/Common/Utils.vb (hunk 1 above)
Comment (davkean) [inline src/Microsoft.VisualStudio.AppDesigner/Common/Utils.vb]: I'd approach this just slightly different - you've done a good job replicating all the different behaviors that they had, but I wouldn't do this - let's pick a policy for the entire project system, and just apply it consistently. ie Either we always call through IsUnrecoverable or we don't (I vote never), either we always Debug.Fail or we don't (I'd probably opt for always just so we can get a feel of how bad this is under debug).

Comment (mavasani) [inline src/Microsoft.VisualStudio.AppDesigner/Common/Utils.vb]: I initially did that - I wasn't sure if cases where there was no Debug.Fail were ones that we meant to swallow exceptions without troubling the developer (especially if it hits too often) OR it was just an oversight from the developer writing that code. I chose the conservative approach for compat, but switching to aggressive approach can also be fine. I can change this method to remove these optional bools, and if it turns out too aggressive we can bring back the flags as and when needed?

Comment (davkean) [inline src/Microsoft.VisualStudio.AppDesigner/Common/Utils.vb]: Yep sounds good.

Comment (mavasani) [inline src/Microsoft.VisualStudio.AppDesigner/Common/Utils.vb]: @davkean - I have pushed another commit addressing your feedback.

**Comment by mavasani** [comment]
Any further feedback?

**Comment by mavasani** [comment]
Ping..

**Comment by davkean** [comment]
:+1:

**Pull request merged**